MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -348,13 +405,24 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputCloseSurfaceModel(vtkMRMLMarkupsT
     modelNode = markupsToModelModuleNode->GetModelNode();
   }
 
-  if(markupsToModelModuleNode->GetButterflySubdivision())
+  if(markupsToModelModuleNode->GetButterflySubdivision() and not OpenSurface)
   {
     vtkSmartPointer< vtkButterflySubdivisionFilter > subdivisionFilter = vtkSmartPointer< vtkButterflySubdivisionFilter >::New();
```

Comment:
Creating and destroying these numerous filters on each update is a lot of unnecessary work. Please have a look at other logic classes how a structure is created for each node that stores all the filters so that they can be reused for each updated instead of rebuilding the entire pipeline from scratch on each update.